src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,129 @@
+package duke;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.Task;
+
+/**
+ * Storage class handles the reading and writing of data
+ * from the local file system.
+ */
+public class Storage {
+    private DateTimeFormatter dateTimeFormat;
+    private File file;
+
+    /**
+     * Public constructor to initialise the storage for the Duii Bot.
+     */
+    public Storage() {
+        this.dateTimeFormat = DateTimeFormatter.ofPattern("yyyy MMM dd HH:mm");
+    }
+
+    /**
+     * Searches the current file system for existing saved files.
+     * Creates a new data directory consisting of the text file to write
+     * into if saved file does not exist.
+     *
+     * @return The abstract representation of the pathname of the created file
+     */
+    public File getfiles() {
+        String home = System.getProperty("user.home");
+        Path path = Paths.get(home, "ip", "data");
+        File directory = new File(path.toString());
+        if (!directory.exists()) {
+            System.out.println("'data' directory not found. Creating the directory...");
+            directory.mkdir();
+            System.out.println("Directory created.");
+        }
+
+        Path filepath = Paths.get(home, "ip", "data", "duke.txt");
+        File file = new File(filepath.toString());
+        if (!file.exists()) {
+            System.out.println("Data file not found. Creating a new file...");
+            try {
+                file.createNewFile();
+                System.out.println("File created.");
+            } catch (IOException e) {
+                System.out.println("IO error occured");
+            }
+        }
+        return file;
+    }
+
+    /**
+     * Writes the saved tasks from previous sessions (if any) into
+     * the local TaskList object.
+     *
+     * @return An ArrayList initialised with the previously saved tasks.
+     * @throws DukeException If an IOException occurs upon reading of file, which is passed on to
+     *                       the main() method of Duke class.
+     */
+    public ArrayList<Task> load() throws DukeException {
+        try {
+            file = this.getfiles();
+            FileReader reader = new FileReader(file);
+            BufferedReader buffreader = new BufferedReader(reader);
+            String text;
+            ArrayList<Task> saved = new ArrayList<>();
```

Comment:
```suggestion
            String text;
            
            ArrayList<Task> saved = new ArrayList<>();
```

Might be good to add a blank line between your variable initializations and your loading logic? The [Layout](https://se-education.org/guides/conventions/java/intermediate.html#layout) section of the course coding standard states that "logical units within a block should be separated by one blank line".